src/sql/sqlops.d.ts

```diff
@@ -1164,7 +1193,7 @@ declare module 'sqlops' {
 	}
 
 	export interface ObjectExplorerProvider extends DataProvider {
-		createNewSession(connInfo: ConnectionInfo): Thenable<ObjectExplorerSessionResponse>;
+		createNewSession?(connInfo: ConnectionInfo): Thenable<ObjectExplorerSessionResponse>;
```

Comment:
I would suggest having 2 separate APIs. Internally, for methods that need to handle both you can use the unified type `ObjectExplorerProvider | ObjectExplorerNodeProvider` to handle this. 

Looking at the other APIs, I really think that sending events for session created / closed is the right call.

For example, you're not awaiting any of the nodeProviders' `closeSession` calls which is incorrect. The reason is you just want to notify them, so it should be a notification instead.

Finally, you could consider `ObjectExplorerProviderBase` and have the 2 separate APIs inherit from that as an alternative solution. Keep the common methods there and split session create / close out from that